dagrunner/utils/logger.py

```diff
@@ -134,24 +149,28 @@ def serve_until_stopped(self, queue_handler=None):
                 self.handle_request()
                 queue_handler.write(self.log_queue)
             abort = self.abort
-        queue_handler.close()
+        if queue_handler:
+            queue_handler.write(self.log_queue)  # Ensure all records are written
+            queue_handler.close()
```

Comment:
This is the change necessary that was before causing intermittent failures.
Essentially this change ensures that we write everything from the log queue before closing things down.